### ISSUE TITLE
Kitana says that Sub-Zero is not enabled?

### ISSUE THREAD

**Issue by krusion** [pannal/Sub-Zero.bundle#753]
So I installed sub zero plugin bundle in plex. I enabled it as the top service under agents in the server, second one is local media assets. Went under settings, and added my login info for a couple of sub repos. Changed the settings and languages to my liking, but it's still not working.

Someone recommended that I should install kitana. So I did that, and in Kitana. Sub-zero says "I'm not enabled". 

Am I missing something?

**Comment by pannal** [comment]
For which agent did you enable it?

**Comment by krusion** [comment]
Hello, thanks for the reply. 

I changed it under 
1. Movies > Plex movie
2. Movies > The movie Database
3. Shows >  TheTVDB
4. Shows > The Movie Database

Selected and moved to the top for all 3. I also tried unselecting and selecting again. Rebooting the server. Tried to even delete the bundle folder, restart server, and put it back in. This kept the settings though. I tried deleting the settings from the database ( not sure if it actually worked or not). And re-add the bundle afterwards. And still didn't work. 

I saw 2 videos that did exactly what I did, but for some reason it didn't work with me but worked with them. 

Thank you for taking the time to help me. I really do appreciate it. Would you prefer to use Slack over this?

**Comment by pannal** [comment]
SZ doesn't work with Plex Movie. You'd have to use the legacy movie client, the new one doesn't call any Python agents.

In order for SZ to work you'd also have to set your libraries to a compatible movie agent (_not_ Plex Movie) and refresh the metadata.

**Comment by krusion** [comment]
Oh I see. 

Is this still doing with the new server or should I rollback? I recently installed this server. But saw mentions of this in older threads. Would appreciate it if you can link me to a tutorial?
 
That being said, I went to look back on the videos I saw and apparently they're using the new user agent, Plex Movie. Obviously they didn't show it in action, so I can't be sure.

https://www.youtube.com/watch?v=YnSRZ2h0yZA

https://www.youtube.com/watch?v=jtMht_QX0_w

**Comment by pannal** [comment]
That's _not_ the new Plex Movie agent. The one that sparked the "Plex Movie (legacy)" agent as the new default landed a few months ago, at max.

You can follow the old agent installation tutorials and SZ's wiki. Just choose the legacy movie agent for your libraries and enable SZ for it.

**Comment by krusion** [comment]
Hello!

You were right! I went into library > manage library > Advanced.

Then changed the Agent to "Plex Movie (Legacy)" instead of "Plex Movie".

Went into agents from the server settings. Fixed that too. Kitana showed
that it's working now. I refreshed Metadata and subtitles are showing up
now!

I'm not sure if using the legacy agent would break something else. If this
is the case, then I'll just manually changed it on a regular basis to get
Sub Zero to work.

I currently don't have any series (TV shows) on here, but it should work,
regardless if I'm using the legacy or new plex movie again, correct? At
least until they change the TV show agent too.


Just tried it with a movie and it worked. I want to download subtitles for
both English and Arabic. Arabic doesn't seem to work for most of the
movies, but it's okay. Better than nothing.


One last question, what's the point of signing into websites? I can make an
account for subscene and OpenSubtitles. But what's the advantage of that if
I don't have a premium account?


Thank you so much for your help. I asked this several times on reddit and
nobody knew what was wrong! I really do appreciate your help and this nice
plugin!

**Comment by pannal** [comment]
Please enlighten reddit, then! Thanks for reporting back.

**Comment by leeaash** [comment]
> That's _not_ the new Plex Movie agent. The one that sparked the "Plex Movie (legacy)" agent has now been the new default, it landed a few months ago, at max.
> 
> You can follow the old agent installation tutorials and SZ's wiki. Just choose the legacy movie agent for your libraries and enable SZ for it.

Really? Is it a joke answer? where did plex said "Plex Movie (legacy)" is a new Plex Movie agent? i only found plex said Plex Movie agent is a new agent to "Plex Movie (legacy)" . On my end, I feel "Plex Movie" is better than "Plex Movie (legacy)" but just only not work with SZ
https://forums.plex.tv/t/introducing-the-new-plex-movie-agent/615989

**Comment by pannal** [comment]
What's your question?

**Comment by yilin101** [comment]
Subzero no compatible the new "Plex Movie" agent.
But new "Plex Movie" is better than "Plex Movie (legacy)"

**Comment by pannal** [comment]
The new Plex Movie agent doesn't support plugins.

**Comment by pannal** [comment]
This is an issue again as Plex has changed the agents endpoint's data recently. Hotfix incoming.

**Comment by pannal** [comment]
Fixed with 2.6.5.3277.

**Comment by frakman1** [comment]
I'm not sure if this is expected behavior or not, but I still get this message after updating the Kitana docker and the Sub-Zero bundle to the latest (0.4.3-2 and 2.6.5.3277 respectively) 

![image](https://github.com/pannal/Sub-Zero.bundle/assets/5826484/45e0ebeb-079d-41c6-8ad6-be847dbc235d)


Note however, that it appears to be working now, in that I am getting subtitles for movies.

For anyone struggling to find the Server Settings Agents page, it's here on the latest Plex:
(I think the instructions [here](https://github.com/pannal/Sub-Zero.bundle/wiki/Installation) are no longer valid for the current Plex UI)

<img width="1533" alt="image" src="https://github.com/pannal/Sub-Zero.bundle/assets/5826484/fa57e52e-1584-47e5-9ce9-e2b8e3e79501">

<img width="1017" alt="image" src="https://github.com/pannal/Sub-Zero.bundle/assets/5826484/f2dc09c0-f3c2-4fce-8819-5035b299b8b3">

@pannal Please correct me if I am doing something wrong or if this is expected. 
Thank you!

**Comment by pannal** [comment]
How are your Libraries configured? Which agents?

**Comment by frakman1** [comment]
I'm not sure if I've had to change that. Do you know where to look?
The Settings page is very confusing and cluttered. I took a look and found this:

<img width="732" alt="image" src="https://github.com/pannal/Sub-Zero.bundle/assets/5826484/cf107f6f-b08a-4924-b2dc-bdf499c9dd41">


Should that be set to 'Plex Movie (Legacy)'?  

Your installation instructions say:

> Additionally, you need to enable the plugin for the library:
> 
> go to Settings -> Server -> Agents -> Movies/TV Shows.
> select the metadata provider you use on your library, e.g.: Freebase (movies) or TheTVDB (series) - Please Note: Any of the newer agents will not work (instead use the older agents "TheTVDB" and "Plex Movie (legacy)"!)
> enable Sub-Zero Subtitles (TV/Movies)
> configure them.
> refresh your library (or individual movies/TV shows)

But there is no such Settings -> Server -> Agents -> Movies/TV Show path. Also, I don't see anywhere to Enable Sub-Zero other than the one from the screenshot I posted yesterday which clearly shows that Sub-Zero is selected for both Movies and Shows.

Can you update the instructions with screenshots please?

**Comment by pannal** [comment]
Exactly the screenshot you're showing. Select "Plex Movie (Legacy)" there.

"Settings -> Server -> Agents -> Movies/TV Show" is now "Settings -> Agents -> Movies/Shows" and yes, those are already properly configured in your case.

You still need to enable Agents for your Libraries that Sub-Zero can actually be used with (TheTVDB and "Plex Movie (legacy)".

**Comment by frakman1** [comment]
Thanks for that.

I changed Movie to use Plex Movie (Legacy)
I changed TV Shows to use TheTVDB 

There is now an icon encouraging me to upgrade those Libraries:

![image](https://github.com/pannal/Sub-Zero.bundle/assets/5826484/df842057-a6ab-46c7-9978-67d6811df6bf)

This is under Settings -> Manage -> Libraries

What does "refresh your library" mean? 
Plex has a `Scan Library Files` button in that page. 
But it also has `Refresh Metadata` and `Analyze` options under the individual Library's options:
It's not clear which one is being called for.

![image](https://github.com/pannal/Sub-Zero.bundle/assets/5826484/11f7cc80-63ac-41ab-97b6-7a843b1572c5)

Anyway, I did both Scan and Refresh Metadata and restarted Plex for good measure. It seems like SubZero is finally happy again:

<img width="1120" alt="image" src="https://github.com/pannal/Sub-Zero.bundle/assets/5826484/8b9a69e2-3732-4d59-9329-c75c36bc7cae">


Thank you for your support.
